### DISCUSSION TITLE
broad-exception-caught / W0718

### DISCUSSION THREAD

**Discussion by DiskCrasher** [microsoft/vscode-pylint#578] [Q&A] [answered]
Is the intent of this warning to persuade the developer to _never_ use `catch Exception`? I can understand if it's the only exception being caught but if it follows one or more other `catch` statements then shouldn't that be enough? I want to say with C# you get a similar error but only if you use `catch Exception` exclusively.

**Comment by karthiknadig** [comment, marked as answer]
@DiskCrasher This question is better suited in the `pylint` repo (https://github.com/pylint-dev/pylint) or their discord (listed here https://pypi.org/project/pylint/). This repo is for the VS Code extension that wraps pylint for UI integrations with VS Code.

**Reply by Pierre-Sassoulas** [reply]
There's also a documentation that explains the intent : https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/broad-exception-caught.html :)

**Reply by DiskCrasher** [reply]
The discussion/argument is now taking place [here](https://github.com/pylint-dev/pylint/issues/9010#issuecomment-2456440965).
